x-pack/test/detection_engine_api_integration/utils/prebuilt_rules/download_prebuilt_rules_from_epr.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ToolingLog } from '@kbn/tooling-log';
+import { openSync, writeSync, unlinkSync, closeSync, statSync } from 'fs';
+import JSON5 from 'json5';
+import path from 'path';
+import { promises as Fsp } from 'fs';
+import { createHash } from 'crypto';
+// @ts-expect-error we have to check types with "allowJs: false" for now, causing this import to fail
+import { REPO_ROOT } from '@kbn/repo-info';
+import chalk from 'chalk';
+
+import Axios from 'axios';
+
+// https://github.com/axios/axios/tree/ffea03453f77a8176c51554d5f6c3c6829294649/lib/adapters
+// @ts-expect-error untyped internal module used to prevent axios from using xhr adapter in tests
+import AxiosHttpAdapter from 'axios/lib/adapters/http';
+
+function tryUnlink(pathToFile: string) {
+  try {
+    unlinkSync(pathToFile);
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      throw error;
+    }
+  }
+}
+
+export const PACKAGE_STORAGE_REGISTRY_URL = 'https://epr.elastic.co';
+
+interface FleetPackage {
+  name: string;
+  version: string;
+  forceAlignStackVersion?: boolean;
+  allowSyncToPrerelease?: boolean;
+}
+
+export const downloadPackageFromFleet = async (bundledPackageDir: string, log: ToolingLog) => {
+  const configFilePath = path.resolve(REPO_ROOT, 'fleet_packages.json');
+  const fleetPackages = await Fsp.readFile(configFilePath, 'utf8');
+
+  const parsedFleetPackages: FleetPackage[] = JSON5.parse(fleetPackages);
+
+  const securityDetectionEnginePackage = parsedFleetPackages.find(
+    (fleetPackage) => fleetPackage.name === 'security_detection_engine'
+  );
+
+  if (!securityDetectionEnginePackage) {
+    throw new Error('Could not find security_detection_engine package');
+  }
+
+  const { name, version } = securityDetectionEnginePackage;
+
+  const packagePath = `${name}-${version}.zip`;
+  const archiveUrl = `${PACKAGE_STORAGE_REGISTRY_URL}/epr/${name}/${name}-${version}.zip`;
+
+  const destination = path.resolve(bundledPackageDir, packagePath);
+
+  try {
+    await downloadToDisk({
+      log,
+      url: archiveUrl,
+      destination,
+      shaChecksum: '',
+      shaAlgorithm: 'sha512',
+      skipChecksumCheck: true,
+      maxAttempts: 3,
+    });
+  } catch (error) {
+    throw new Error(`Failed to download bundled package archive ${packagePath}: ${error.message}`);
+  }
+
+  return { version };
+};
+
+interface DownloadToDiskOptions {
+  log: ToolingLog;
+  url: string;
+  destination: string;
+  shaChecksum: string;
+  shaAlgorithm: string;
+  maxAttempts?: number;
+  retryDelaySecMultiplier?: number;
+  skipChecksumCheck?: boolean;
+}
+export async function downloadToDisk({
```

Comment:
The implementation of this method appears quite complex. I would recommend avoiding the use of such intricate logic in tests.

Reply:
I actually just copied these methods from the build process for Kibana, since I cannot import them into our folder. I agree that these would become a pain to maintain. 
I guess this depends of how do we want to test the installation on air-gapped envs. If we do decide against acutally downloading the package from EPR during testing, we can get rid of all this download logic.